### PR TITLE
Add get_targets and get_quotas to Workspace

### PR DESCRIPTION
**Pull request by guenp**
Add `Workspace.get_targets` and `Workspace.get_job_quotas`
Depends on #111

### CODE REVIEW

File: azure-quantum/azure/quantum/workspace.py [hunk 1]
```diff
@@ -257,6 +257,27 @@ def list_jobs(
...
+            ] for provider in self._client.providers.get_status()
+        }
+
+    def get_job_quotas(self) -> List[Dict[str, Any]]:
+        """Get a list of job quotas for the given workspace.
+
+        :return: Job quotas
+        :rtype: List[Dict[str, Any]]
+        """
+        return [q.as_dict() for q in self._client.quotas.list()]
```
Comment (anpaz) [inline azure-quantum/azure/quantum/workspace.py:279]: `self._client.quotas` should be wrapped in its own `_get_quotas_client`

**Review by anpaz — APPROVED** [review]
Just waiting on release build to make sure Live tests work well.

File: azure-quantum/azure/quantum/workspace.py [hunk 2]
```diff
@@ -257,6 +257,27 @@ def list_jobs(
...
+        :return: Targets, keyed by provider IDs
+        :rtype: Dict[str, List[str]]
+        """
+        return {
+            provider.id: [
+                target.id for target in provider.targets
+            ] for provider in self._client.providers.get_status()
+        }
+
+    def get_job_quotas(self) -> List[Dict[str, Any]]:
```
Comment (anpaz) [inline azure-quantum/azure/quantum/workspace.py:273]: job_quotas, seems to refer as the quotas of a single job, maybe just keep it `quotas`?

File: azure-quantum/azure/quantum/workspace.py (hunk 1 above)
Comment (guenp) [reply azure-quantum/azure/quantum/workspace.py:279]: Sure, good idea to keep it consist

**Pull request merged**